app/models/presentation.rb

```diff
@@ -26,6 +26,7 @@ def person_or_presenter
   end
 
   def associated_event_has_available_time
+    return unless duration
```

Comment:
Agree with item 1 and have updated accordingly (another option could be if `errors.any?`, but i like that this is more specific to what this custom validation requires)

WRT item 2, as discussed, there is already a duration validation in the presentation model but it seems that rails will run all validations even if one fails, so if you have a custom validation that is dependent on an attribute being valid, you will have to test explicity for this in your custom validation. Ref http://stackoverflow.com/questions/5966055/controlling-the-order-of-rails-validations